cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaReconciler.java

```diff
@@ -810,12 +815,41 @@ protected Future<Void> podDisruptionBudgetV1Beta1() {
     }
 
     /**
-     * Prepares annotations for Kafka pods which are known only in the KafkaAssemblyOperator level. These are later
-     * passed to KafkaCluster where there are used when creating the Pod definitions.
+     * Prepares annotations for Kafka pods within a StrimziPodSet which are known only in the KafkaAssemblyOperator level.
+     * These are later passed to KafkaCluster where there are used when creating the Pod definitions.
      *
-     * @return  Map with Pod annotations
+     * @return Map with Pod annotations
      */
-    private Map<String, String> kafkaPodAnnotations(int brokerId, boolean storageAnnotation)    {
+    private Map<String, String> podSetPodAnnotations(int brokerId) {
+        Map<String, String> podAnnotations = commonKafkaPodAnnotations(brokerId);
+
+        // Annotation of broker certificate hash
+        podAnnotations.put(KafkaCluster.ANNO_STRIMZI_SERVER_CERT_HASH, kafkaServerCertificateHash.get(brokerId));
+
+        return podAnnotations;
+    }
+
+    /**
+     * Prepares annotations for Kafka pods within a StatefulSet which are known only in the KafkaAssemblyOperator level.
+     * These are later passed to KafkaCluster where there are used when creating the Pod definitions.
+     *
+     * @return Map with Pod annotations
+     */
+    private Map<String, String> statefulSetPodAnnotations() {
+        Map<String, String> podAnnotations = commonKafkaPodAnnotations(sharedConfigurationId);
+
+        // Storage annotation on Pods is used only for StatefulSets
+        podAnnotations.put(ANNO_STRIMZI_IO_STORAGE, ModelUtils.encodeStorageToJson(kafka.getStorage()));
+
+        return podAnnotations;
+    }
+
+    /**
+     * Prepares the Kafka pods' annotations that are common between StatefulSets and StrimziPodSets.
+     *
+     * @return Map with Pod annotations
+     */
```

Comment:
A nit ... but maybe you can add the missing param description to the Javadoc?

Reply:
Done